src/main/scala/me/jeffmay/neo4j/client/ws/WSNeo4jClient.scala

```diff
@@ -0,0 +1,208 @@
+import play.api.libs.ws.{WSAuthScheme, WSClient, WSRequest, WSResponse}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future, blocking}
+import scala.util.Try
+
+class WSNeo4jClient(
+  wsClient: WSClient,
+  val config: Neo4jClientConfig,
+  implicit private val scheduler: Scheduler,
+  implicit private val executionContext: ExecutionContext
+) extends Neo4jClient with Proxy {
+  client =>
+
+  import WSNeo4jClient._
+  import WSRequestWithException._
+
+  override def self: Any = (wsClient, config)
+
+  def copy(
+    ws: WSClient = this.wsClient,
+    config: Neo4jClientConfig = this.config,
+    scheduler: Scheduler = this.scheduler,
+    executionContext: ExecutionContext = this.executionContext
+  ) = {
+    new WSNeo4jClient(wsClient, config, scheduler, executionContext)
+  }
+
+  lazy val ws: WSClient = new ProxyWSClient(wsClient) {
+    override def url(url: String): WSRequest = {
+      new TimeoutWSRequest(wsClient.url(url), config.timeout)
+    }
+  }
+
+  override def withAutoInitialize(autoAuthenticate: Boolean): Neo4jClient =
+    this.copy(config = config.copy(autoInitialize = autoAuthenticate))
+
+  override def withCredentials(username: String, password: String): Neo4jClient =
+    this.copy(config = config.copy(credentials = Neo4jBasicAuth(username, password)))
+
+  override def withBaseUrl(baseUrl: String): Neo4jClient =
+    this.copy(config = config.copy(baseUrl = baseUrl))
+
+  override def withStatsIncludedByDefault(includeStatsByDefault: Boolean): Neo4jClient =
+    this.copy(config = config.copy(includeStatsByDefault = includeStatsByDefault))
+
+  override def withTimeout(timeout: FiniteDuration): Neo4jClient =
+    this.copy(config = config.copy(timeout = timeout))
+
+  protected def whenInitialized[T](action: => Future[T]): Future[T] = {
+    if (config.autoInitialize) {
+      WSNeo4jClient.withVerifiedCredentials(withCredentials("neo4j", "neo4j"), config.credentials)(action)
+    }
+    else action
+  }
+
+  protected def http(path: String): WSRequest = {
+    require(!path.isEmpty, "path cannot be empty")
+    require(path.charAt(0) == '/', "path must be an absolute path starting with '/'")
+    import config.credentials.{password, username}
+    ws.url(config.baseUrl + path).withAuth(username, password, WSAuthScheme.BASIC)
+  }
+
+  override def passwordChangeRequired(): Future[Boolean] = {
+    val request = http(s"/user/${config.credentials.username}")
+    whenInitialized {
+      request.getAndCheckStatus(Set(200)).map { resp =>
+        (resp.json \ "password_change_required").asOpt[Boolean] contains true
+      }
+    }
+  }
+
+  override def changePassword(newPassword: String): Future[Unit] = {
+    val request = http(s"/user/${config.credentials.username}/password")
+    request.postAndCheckStatus(Some(Json.obj("password" -> newPassword)), Set(200))
+      .map(_ => ())
+  }
+
+  override def openTxn(statements: Seq[CypherStatement]): Future[OpenedTxnResponse] = {
```

Comment:
Also, `Future` is a bad API for supporting streams on a batch of queries like this. I might want to name it a bit more awkwardly until I figure out how to do streaming.

Ideally, I'd like to boil all operations down into a single batch / streaming API that I can then call for single results (`Future[X]`s), pre-evaluated batch results (`Future[Seq[X]]`) and streams (both `Enumerator[X]` and `Publisher[X]`). I'm guessing I will eventually use Akka Streams to produce the `Publisher` and then convert the publisher into an `Enumerator` for better Play support, since Play controllers use the Play Iteratees (even in Play 2.4). But they claim they will move over to supporting `Publisher`s and support Iteratees for backwards compatibility in some future release, but 2.5.x doesn't seem to have made the switch, so I don't know when that is coming.